lib/flight_scheduler/job_registry.rb

```diff
@@ -47,12 +47,18 @@ def initialize
       @jobs = Concurrent::Hash.new
     end
 
-    def add_job(job_id, job)
+    def each_job
+      @jobs.each do |_, job|
+        yield job[:job] if block_given?
+      end
+    end
+
+    def add_job(job_id, job, start_timeout: true)
```

Comment:
We call this method in only two locations.  The first location is from `MessageProcessor` and the second whilst loading the persisted jobs.  That half the clients of this method want to start the time and the other half don't, is a strong indication that adding jobs to the registry and starting the timer are orthogonal concerns.  We would be better off moving the time starting from the registry to `MessageProcessor`.

Reply:
https://github.com/openflighthpc/flight-scheduler-daemon/pull/29/commits/24bf6db61eefd75c22c88eee9001c14d46f70b1c